src/xma/src/xmaapi/xmaapi.cpp

```diff
@@ -267,6 +271,7 @@ void xma_thread2(uint32_t hw_dev_index) {
             xrt_device_obj.get_handle()->exec_wait(100);
         }
 
+        std::lock_guard<std::mutex> lock(g_xma_singleton->m_mutex);
```

Comment:
can't we move this to before line number 267?

Reply:
Same reasoning as above 👆